### PR TITLE
Convert apis

### PR DESCRIPTION
**Pull request by b1llow**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://cutter.re/docs/contributing/code/getting-started.html) to this repository
- [x] I made sure to follow the project's [coding style](https://cutter.re/docs/contributing/code/development-guidelines.html)
- [ ] I've updated the [documentation](https://cutter.re/docs/user-docs.html) with the relevant information (if needed)

**Detailed description**

**Test plan (required)**

**Closing issues**

correspond https://github.com/rizinorg/rizin/pull/2630

Partially addresses #2666

### CODE REVIEW

**Review by XVilka — APPROVED** [review]
Looks good enough from the porting commands to the API point of view. However, the Cutter code doesn't handle errors at all. For example, these debug steps or reopening functions can easily fail and it's not handled at all. Of course, it wasn't handled even before, when commands were used. But this is outside of scope of this work and should be done separately. I will create a new issue to track that.

File: src/core/Cutter.cpp [hunk 1]
```diff
@@ -4252,17 +4324,22 @@ void CutterCore::setWriteMode(bool enabled)
...
-        // Change from write-mode to read-only
+    if (enabled) {
+        if (!writeModeState) {
+            rz_core_io_file_reopen(core, core->io->desc->fd, RZ_PERM_RW);
+        }
     } else {
-        cmdRaw("oo");
+        // Change from write-mode to read-only
+        rz_core_io_file_open(core, core->io->desc->fd);
     }
```
Comment (ret2libc) [inline src/core/Cutter.cpp:4329]: This seems slightly different from the old behaviour. Is that on purpose?

Comment (b1llow) [reply src/core/Cutter.cpp:4329]: Yes.

File: src/core/Cutter.cpp [hunk 2]
```diff
@@ -802,7 +808,12 @@ void CutterCore::jmpReverse(RVA addr)
 
 void CutterCore::editBytes(RVA addr, const QString &bytes)
 {
-    cmdRawAt(QString("wx %1").arg(bytes), addr);
+    applyAtSeek(
```
Comment (p371jah) [inline src/core/Cutter.cpp]: `applyAtSeek` is redudant here since you use API.

File: src/widgets/HexWidget.cpp [hunk 3]
```diff
@@ -894,9 +897,22 @@ void HexWidget::w_duplFromOffset()
...
+        return;
+    }
+    int n = rz_io_nread_at(core->io, src, buf, len);
+    if (n < 0) {
+        free(buf);
+        return;
+    }
+    rz_core_write_at(core, getLocationAddress(), buf, n);
+    free(buf);
+
```
Comment (ret2libc) [inline src/widgets/HexWidget.cpp]: let me add an API for this.

Comment (ret2libc) [reply src/widgets/HexWidget.cpp]: https://github.com/rizinorg/rizin/pull/2633

**Review by XVilka — APPROVED** [review]

**Review by ret2libc — APPROVED** [review]

File: src/core/Cutter.cpp [hunk 4]
```diff
@@ -656,7 +656,13 @@ bool CutterCore::loadFile(QString path, ut64 baddr, ut64 mapaddr, int perms, int
     }
 
     if (perms & RZ_PERM_W) {
-        rz_core_cmd0(core, "omfg+w");
+        RzPVector *maps = rz_io_maps(core->io);
+        void **it;
+        rz_pvector_foreach(maps, it)
+        {
+            auto *map = static_cast<RzIOMap *>(*it);
```
Comment (thestr4ng3r) [inline src/core/Cutter.cpp]:
```suggestion
            auto *map = reinterpret_cast<RzIOMap *>(*it);
```

or even more elegantly:
```c++
for (auto map : CutterPVector<RzIOMap>(maps)) {
    // ...
```

File: src/widgets/HexWidget.cpp [hunk 5]
```diff
@@ -813,24 +811,26 @@ void HexWidget::w_writeZeros()
...
+    bool ok = false;
+    int len = QInputDialog::getInt(this, tr("Write zeros"), tr("Number of zeros:"), size, 1,
+                                   0x7FFFFFFF, 1, &ok);
+    if (!ok) {
+        return;
     }
+
+    RzCoreLocked core(Core());
+    auto *buf = (uint8_t *)calloc(len, sizeof(uint8_t));
+    rz_core_write_at(core, getLocationAddress(), buf, len);
```
Comment (thestr4ng3r) [inline src/widgets/HexWidget.cpp]: Better limit the scope of `RzCoreLocked` further, like:
```c++
// ...
{
    RzCoreLocked core(Core());
    auto *buf = (uint8_t *)calloc(len, sizeof(uint8_t));
    rz_core_write_at(core, getLocationAddress(), buf, len);
    free(buf);
}
refresh();
```
same applies to the cases below.

**Review by thestr4ng3r — CHANGES_REQUESTED** [review]
overall looks good

**Review by thestr4ng3r — APPROVED** [review]

**Pull request merged**